src/ansys/geometry/core/designer/design.py

```diff
@@ -679,6 +690,51 @@ def add_beam_circular_profile(
 
         return self._beam_profiles[profile.name]
 
+    @protect_grpc
+    @min_backend_version(25, 1, 0)
+    def get_all_parameters(self):
+        """Get parameters for the design.
+
+        Returns
+        -------
+        List[DrivingDimension]
+            List of parameters for the design.
+        """
+        response = self._parameters_stub.GetAll(GetAllRequest())
+        return [Parameter._from_proto(dimension) for dimension in response.driving_dimensions]
+
+    @protect_grpc
+    @check_input_types
+    @min_backend_version(25, 1, 0)
+    def set_parameter(self, dimension: Parameter) -> ParameterUpdateStatus:
+        """Update a parameter of the design.
+
+        Parameters
+        ----------
+        dimension : Parameter
+            Parameter to set.
+
+        Returns
+        -------
+        ParameterUpdateStatus
+            Enum containing the status of the update operation.
+        """
+        request = UpdateRequest(driving_dimension=Parameter._to_proto(dimension))
+        response = self._parameters_stub.UpdateParameter(request)
+        status = response.status
+
+        status_mapping = {
+            UpdateStatus.SUCCESS: ParameterUpdateStatus.SUCCESS,
+            UpdateStatus.FAILURE: ParameterUpdateStatus.FAILURE,
+            UpdateStatus.CONSTRAINED_PARAMETERS: ParameterUpdateStatus.CONSTRAINED_PARAMETERS,
+        }
+
+        # Update the design in place. This method is expensive,
+        # consider finding a more efficient approach.
+        self._update_design_inplace()
+
+        return status_mapping.get(status, ParameterUpdateStatus.UNKNOWN)
+
```

Comment:
I would do this conversion and mapping as a private static method of the ``ParameterUpdateStatus`` class. Have a look at other examples like: https://github.com/ansys/pyansys-geometry/blob/76116c829c7007a4ba64938b4ba3f91bf1667f78/src/ansys/geometry/core/connection/backend.py#L38-L70

In that case the staticmethod is public, but I would make it "private" with a leading underscore for this purpose. That way we can reuse this method in other locations if needed. Otherwise we would need to copy-paste this logic

Reply:
make sense, done.